extensions/target-specific/chromium/gfx-type-helpers/gfx-type-helpers.js

```diff
@@ -54,4 +54,51 @@ Loader.OnLoad(function() {
         UserEditableFunctions.Create((scrollOffset) => Promise.all([scrollOffset.f("x_").val(), scrollOffset.f("y_").val()])
             .thenAll((first, second) => `{${first}, ${second}}`))
     );
+
+    DbgObject.AddTypeDescription(
+        DbgObjectType("viz_common", "gfx::Rect"),
```

Comment:
I'd like to get Sanket's take on how we should rationalize the process model here - ideally these new types would be supported for all processes (I suspect there is usage of these types in the browser process as well).

In the short-term though, can we use the GpuProcessType wrapper so that this works on non-component builds as well?

Reply:
Yes, we should definitely wrap all of these with Chromium.GpuProcessType so that both component and non-component builds are supported.

In terms of supporting types in multiple processes...even if the same type exists in mutliple process, they are both technically distinct types. So I think we need a declarative way to add the same extended field/type description/array field on distinct types. This way we can share code and have the same helpers appear across processes. Perhaps, AddExtendedField/AddTypeDescription/AddArrayField could take an array of input types?

If this is not an immediate blocker, we can address this separately. Feel free to file an issue and just add the GpuProcessType wrapper for now.

Reply:
You should still replace DbgObjectType with Chromium.GpuProcessType for all of these fields, and validate that they work with non-component builds.

Reply:
I think I have replaced DbgObjectType in revision 2 ? I missed one in   CompositorFrame.js updating that now.

Reply:
non component builds build with symbol level != 2, in such a build I cannot set break points on the compositor frame sink. So at the moment I cannot test this, though I have updated the process names like Daniel suggested.

Reply:
release (dev/canary) of msedge have symbol_level=2. It may be the case that that particular symbol is inlined, but you should still be able to test

Reply:
E:\edgees\src>gn gen out/NoComp --args="is_component_build = false is_debug = true use_jumbo_build = false enable_nacl = false is_win_fastlink = true use_goma = true symbol_level = 2 goma_dir =\"%GOMACLIENTDIR%\""
ERROR at //build/config/compiler/compiler.gni:263:3: Assertion failed.
  assert(symbol_level != 2 || current_toolchain != default_toolchain ||
  ^-----
Can't do non-component debug builds at symbol_level=2

Reply:
Right, but we want these things to work with release builds as well.

Is there anything debug specific to your extension?

Reply:
Let me try release builds again, I had trouble getting a pointer to a compositor frame in release builds before.